avocado/utils/partition.py

```diff
@@ -0,0 +1,299 @@
+        :param loop_size: Size of loopback device (in MB). Defaults to 0.
+        :param mountpoint: Where the partition to be mounted to.
+        """
+        self.device = device
+        self.loop = loop_size
+        self.fstype = None
+        self.mountpoint = mountpoint
+        self.mkfs_flags = ''
+        self.mount_options = None
+        if self.loop:
+            process.run('dd if=/dev/zero of=%s bs=1M count=%d'
+                        % (device, self.loop))
+
+    def __repr__(self):
+        return '<Partition: %s>' % self.device
+
+    @staticmethod
+    def list_mount_devices():
+        """
+        Lists mounted file systems and swap on devices.
+        """
+        # list mounted file systems
+        devices = [line.split()[0]
+                   for line in process.system_output('mount').splitlines()]
+        # list mounted swap devices
+        swaps = process.system_output('swapon -s').splitlines()
+        devices.extend([line.split()[0] for line in swaps
+                        if line.startswith('/')])
+        return devices
+
+    @staticmethod
+    def list_mount_points():
+        """
+        Lists the mount points.
+        """
+        return [line.split()[2]
+                for line in process.system_output('mount').splitlines()]
+
+    def get_mountpoint(self, filename=None):
+        """
+        Find the mount point of this partition object.
+
+        :param filename: where to look for the mounted partitions information
+                (default None which means it will search /proc/mounts and/or
+                /etc/mtab)
+
+        :return: a string with the mount point of the partition or None if not
+                mounted
+        """
+        # Try to match this device/mountpoint
+        if filename:
+            for line in open(filename):
+                parts = line.split()
+                if parts[0] == self.device or parts[1] == self.mountpoint:
+                    return parts[1]    # The mountpoint where it's mounted
+            return None
+
+        # no specific file given, look in /proc/mounts
+        res = self.get_mountpoint(filename='/proc/mounts')
+        if not res:
+            # sometimes the root partition is reported as /dev/root in
+            # /proc/mounts in this case, try /etc/mtab
+            res = self.get_mountpoint(filename='/etc/mtab')
+
+            if res != '/':
+                res = None
+        return res
+
+    def mkfs(self, fstype=None, args=''):
+        """
+        Format a partition to filesystem type
+
+        :param fstype: the filesystem type, e.g.. "ext3", "ext2"
+        :param args: arguments to be passed to mkfs command.
+        """
+
+        if self.device in self.list_mount_devices():
+            raise PartitionError(self, 'Unable to format mounted device')
+
+        if not fstype:
+            if self.fstype:
+                fstype = self.fstype
+            else:
+                fstype = 'ext2'
```

Comment:
Well I think we can improve it later, this is just backport from autotest and it is working fine...